test/plugin_test.py.in

```diff
@@ -1306,6 +1307,78 @@ class TestPlugin(unittest.TestCase):
 
                     self.c.access_group_delete(ag)
 
+
+    def test_volume_delete_with_replication(self):
+        for s in self.systems:
+            cap = self.c.capabilities(s)
+            if supported(cap, [Cap.VOLUME_REPLICATE_CLONE,
+                               Cap.VOLUME_CREATE,
+                               Cap.VOLUME_DELETE]):
+                pool = self._get_pool_by_usage(s.id,
+                                               lsm.Pool.ELEMENT_TYPE_VOLUME)
+                if pool:
+                    vol_size = self._min_size()
+
+                    vol = self.c.volume_create(
+                        pool, rs('v'), vol_size,
+                        lsm.Volume.PROVISION_DEFAULT)[1]
+
+                    if vol:
+                        got_exception = False
+                        clone_vol = self.c.volume_replicate(
+                            pool, lsm.Volume.REPLICATE_CLONE, vol,
+                            rs('test_vd_with_repo_'))[1]
+
+                        try:
+                            self.c.volume_delete(vol)
+                        except LsmError as le:
+                            if le.code == lsm.ErrorNumber.HAS_CHILD_DEPENDENCY:
+                                got_exception = True
+                            self.assertTrue(
+                                le.code == lsm.ErrorNumber.HAS_CHILD_DEPENDENCY)
+
+                        self.assertTrue(got_exception)
+
+                        # Clean up
+                        self.c.volume_delete(clone_vol)
+                        self.c.volume_delete(vol)
+            else:
+                self._skip_current_test(
+                    "Skip test: current system does not support "
+                    "volume_create() or volume_replicate() with CLONE "
+                    "or volume_delete()")
+
+
+    def test_fs_delete_with_clone(self):
+        for s in self.systems:
+            cap = self.c.capabilities(s)
+
+            if supported(cap, [Cap.FS_CREATE, Cap.FS_CLONE, Cap.FS_DELETE]):
+                fs, pool = self._fs_create(s.id)
+                self.assertTrue(fs is not None)
+                fs_child = self.c.fs_clone(fs, rs('fs_c_'))[1]
+                if self.c.fs_child_dependency(fs, None):
```

Comment:
Need to add a check here to see if `fs_child_dependency` is supported before calling.